kernel/src/fs/procfs/template/dir.rs

```diff
@@ -183,10 +186,52 @@ impl<D: DirOps + 'static> Inode for ProcDir<D> {
     }
 }
 
-pub trait DirOps: Sync + Send {
-    fn lookup_child(&self, this_ptr: Weak<dyn Inode>, name: &str) -> Result<Arc<dyn Inode>> {
-        Err(Error::new(Errno::ENOENT))
+pub trait DirOps: Sync + Send + Sized {
+    fn lookup_child(&self, dir: &ProcDir<Self>, name: &str) -> Result<Arc<dyn Inode>>;
+
+    fn populate_children<'a>(
+        &self,
+        dir: &'a ProcDir<Self>,
+    ) -> RwMutexUpgradeableGuard<'a, SlotVec<(String, Arc<dyn Inode>)>>;
+
+    #[must_use]
+    fn validate_child(&self, _child: &dyn Inode) -> bool {
+        true
+    }
+}
+
+pub fn lookup_child_from_table<T, F>(
+    name: &str,
+    cached_children: &mut SlotVec<(String, Arc<dyn Inode>)>,
+    table: &[(&str, T)],
+    func: F,
+) -> Option<Arc<dyn Inode>>
+where
+    T: Copy,
+    F: FnOnce(T) -> Arc<dyn Inode>,
+{
+    for (child_name, child_data) in table.iter() {
+        if *child_name == name {
+            return Some(
+                cached_children
+                    .put_entry_if_not_found(name, || (func)(*child_data))
+                    .clone(),
+            );
+        }
     }
 
-    fn populate_children(&self, this_ptr: Weak<dyn Inode>) {}
+    None
+}
+
+pub fn populate_children_from_table<T, F>(
+    cached_children: &mut SlotVec<(String, Arc<dyn Inode>)>,
+    table: &'static [(&'static str, T)],
```

Comment:
Can we remove the `'static` lifetimes here to keep consistent with `lookup_child_from_table`?